contracts/light-clients/AttestorLightClient.sol

```diff
@@ -0,0 +1,216 @@
+    /// @param initialTimestampSeconds The initial timestamp in seconds for the initial height
+    /// @param roleManager Address that will administer roles and be allowed to submit proofs; if zero, anyone can
+    /// submit
+    constructor(
+        address[] memory attestorAddresses,
+        uint8 minRequiredSigs,
+        uint64 initialHeight,
+        uint64 initialTimestampSeconds,
+        address roleManager
+    ) {
+        if (attestorAddresses.length == 0) revert NoAttestors();
+        if (minRequiredSigs == 0 || minRequiredSigs > attestorAddresses.length) {
+            revert BadQuorum(minRequiredSigs, attestorAddresses.length);
+        }
+
+        clientState = IAttestorLightClientMsgs.ClientState({
+            attestorAddresses: attestorAddresses,
+            minRequiredSigs: minRequiredSigs,
+            latestHeight: initialHeight,
+            isFrozen: false
+        });
+
+        for (uint256 i = 0; i < attestorAddresses.length; ++i) {
+            _isAttestor[attestorAddresses[i]] = true;
+        }
+
+        _consensusTimestampAtHeight[initialHeight] = initialTimestampSeconds;
+
+        if (roleManager == address(0)) {
+            _grantRole(PROOF_SUBMITTER_ROLE, address(0)); // allow anyone
+        } else {
+            _grantRole(DEFAULT_ADMIN_ROLE, roleManager);
+            _grantRole(PROOF_SUBMITTER_ROLE, roleManager);
+        }
+    }
+
+    /// @inheritdoc ILightClient
+    function getClientState() external view returns (bytes memory) {
+        return abi.encode(clientState);
+    }
+
+    /// @inheritdoc IAttestorLightClient
+    function getAttestorSet() external view returns (address[] memory attestorAddresses, uint8 minRequiredSigs) {
+        return (clientState.attestorAddresses, clientState.minRequiredSigs);
+    }
+
+    /// @inheritdoc IAttestorLightClient
+    function getConsensusTimestamp(uint64 revisionHeight) external view returns (uint64) {
+        return _consensusTimestampAtHeight[revisionHeight];
+    }
+
+    /// @inheritdoc ILightClient
+    function updateClient(bytes calldata updateMsg)
+        external
+        notFrozen
+        onlyProofSubmitter
+        returns (ILightClientMsgs.UpdateResult)
+    {
+        IAttestorMsgs.AttestationProof memory proof = abi.decode(updateMsg, (IAttestorMsgs.AttestationProof));
+
+        bytes32 digest = sha256(proof.attestationData);
+        _verifySignatures(digest, proof.signatures);
+
+        IAttestorMsgs.StateAttestation memory state =
+            abi.decode(proof.attestationData, (IAttestorMsgs.StateAttestation));
+
+        // Check if height already exists, if it does, check if the timestamp is the same, otherwise revert
+        if (_consensusTimestampAtHeight[state.height] != 0) {
+            if (_consensusTimestampAtHeight[state.height] != state.timestamp) {
+                revert ConflictingTimestamp(state.height, _consensusTimestampAtHeight[state.height], state.timestamp);
```

Comment:
This should not revert. It should freeze the light client and return `UpdateResult.Misbehaviour`